### ISSUE TITLE
[JAVA, JS] A class with no members for model of type string

### ISSUE THREAD

**Issue by danadam** [swagger-api/swagger-codegen#3162]
##### Description

I have a model of type string (with maxLength but it doesn't affect anything). Java class generated for it has no members at all:
```java
public class Foo   {
  

  
  @Override
  public boolean equals(java.lang.Object o) {
```

##### Swagger-codegen version

``cc5e79cd0d19 (2016-06-17 19:36:24 +0800) wing328 - move server stub doc to wiki``

##### Swagger declaration file content or url

```yaml
swagger: "2.0"
info:
    version: "0.1.0"
    title: bla
host: example.com
paths:
    /example:
        get:
            operationId: bla
            parameters:
            -   name: MyParam
                in: body
                schema:
                    type: object
                    properties:
                        bar:
                            $ref: '#/definitions/foo'
            responses:
                default:
                    description: bla

definitions:
    foo:
        type: string
        maxLength: 32
```

##### Command line used for generation

``java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -i bug.yaml -l java -o bla``

**Comment by wing328** [comment]
`foo` is missing the `properties`, e.g. https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/test/resources/2_0/petstore.yaml#L613

**Comment by danadam** [comment]
True, because it is of type ``string`` and ``properties`` is a valid keyword for type ``object`` (according to [JSON Schema Validation](http://json-schema.org/latest/json-schema-validation.html) document).

This is a valid spec file, at least according to [swagger editor](http://editor.swagger.io), so I would expect that:

* either ``MyParam`` class has a ``foo`` member of type ``String`` (and there is no ``Foo`` class),
* or ``MyParam`` class has a ``foo`` member of type ``Foo`` and ``Foo`` class has a ``value`` member (or similar) of type ``String``.

Right now ``MyParam`` class has a ``foo`` member of type ``Foo`` and ``Foo`` class has no members.

Or is there a rule somewhere that models in ``definitions`` can be only of type ``object``? I'm just starting with JSON and swagger so it is entirely possible that I missed something :-)

**Comment by daisymoon** [comment]
Yes, I have the same behaviour using this [spec](https://watson-api-explorer.mybluemix.net/listings/visual-recognition-v3.json).
Since `type` doesn´t show in the json generated for the models (`java -DdebugModels -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -i https://watson-api-explorer.mybluemix.net/listings/visual-recognition-v3.json -l csharp`) what I did was this (in a Codegen class I´m building for my own language):

```
public Map<String, Object> postProcessModels(Map<String, Object> objs) {
 ...
 if (!cm.hasVars) {   
   String model = cm.modelJson.replaceAll("(\\r|\\n)", "");
   ObjectMapper mapper = new ObjectMapper();
   try {
      JsonNode json = mapper.readTree(model);
      List<String> type = json.findValuesAsText("type");
      if (type != null) {
           CodegenProperty propType = new CodegenProperty();
            propType.name = cm.classname;
            propType.datatype = getDataType(type.get(0));
            propType.example = extraProperties(propType.datatype, type.get(0));
            propType.datatypeWithEnum = propType.datatype;
            cm.allVars.add(propType);
      }
  } catch (Exception e) {
       System.out.println(e.getMessage());
  }
 }
}
```

This way I get the second solution: or `MyParam` class has a `foo `member of type `Foo` and `Foo` class has a value member (or similar) of type `String`
I think it would be a good idea to consider this possibility of having a model with a `type` and no `properties`

**Comment by wing328** [comment]
@danadam I believe you want to reference the property to something you defined so that you can reuse it or change it more easily later for multiple attributes. That's not supported at the moment.

As a workaround, please define the property without using $ref.

**Comment by danadam** [comment]
My use case is a primitive with limits used in many places, e.g., country code (2 character string). I was just hoping that it would be possible to create a simpler API. With the following schema:
```javascript
{
  "type": "object",
  "properties": {
    "nationality": {
      "$ref": "#/definitions/countryCode"
    },
    "countryOfResidence": {
      "$ref": "#/definitions/countryCode"
    }
  },
  "definitions": {
    "countryCode": {
      "type": "string",
      "maxLength": 2
    }
  }
}
```
The client would be able to send simply:
```javascript
{
  "nationality": "us",
  "countryOfResidence": "gb"
}
```
Now I will use this schema instead:
```javascript
{
  "type": "object",
  "properties": {
    "nationality": {
      "$ref": "#/definitions/countryCode"
    },
    "countryOfResidence": {
      "$ref": "#/definitions/countryCode"
    }
  },
  "definitions": {
    "countryCode": {
      "type": "object",
      "properties": {
        "value": {
          "type": "string",
          "maxLength": 2
        }
      }
    }
  }
}
```
So the client will have to send:
```javascript
{
  "nationality": {
    "value": "us"
  },
  "countryOfResidence": {
    "value": "gb"
  }
}
```

**Comment by wing328** [comment]
Well you can simply define a model with 2 properties: `countryOfResidence` and `nationality` for the following JSON payload:
```
{
  "nationality": "us",
  "countryOfResidence": "gb"
}
```
e.g. https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/test/resources/2_0/petstore.yaml#L612
